Source/WebKit/Platform/cocoa/ExtensionCapability.h

```diff
@@ -31,21 +31,34 @@
 #include <wtf/RetainPtr.h>
 
 OBJC_CLASS BEProcessCapability;
+OBJC_CLASS _SECapability;
 
 namespace WebKit {
 
+using PlatformCapability = std::variant<RetainPtr<BEProcessCapability>, RetainPtr<_SECapability>>;
+
 class ExtensionCapability {
 public:
     virtual ~ExtensionCapability() = default;
     virtual String environmentIdentifier() const = 0;
-    BEProcessCapability *platformCapability() const { return m_platformCapability.get(); }
+    PlatformCapability platformCapability() const { return m_platformCapability; }
+
+    bool hasPlatformCapability() const { return platformCapabilityIsValid(m_platformCapability); }
+
+    static bool platformCapabilityIsValid(const PlatformCapability& platformCapability)
+    {
+        WTF::switchOn(platformCapability, [&] (auto& capability) {
+            return !!capability;
+        });
+        return false;
+    }
 
 protected:
     ExtensionCapability() = default;
-    void setPlatformCapability(BEProcessCapability *capability) { m_platformCapability = capability; }
+    void setPlatformCapability(PlatformCapability capability) { m_platformCapability = capability; }
```

Comment:
`PlatformCapability&&` or `const PlatformCapability&`. Also `std::move()` if you go with the r-value reference.